### PR TITLE
Support build-time dependencies from other storages

### LINKED ISSUE #3446: Pulp - runtime / build-time dependencies

### PR DESCRIPTION
**Pull request by FrostyX**
Fix #3446

When a Copr defines a build-time dependency on `copr://foo/bar` and the `foo/bar` project uses Pulp storage, the repository baseurl is now generated correctly.

While the change itself was very simple, I needed to move some code around to avoid circular dependencies.

### CODE REVIEW

**Comment by FrostyX** [comment]
I finally tested this on STG and it works fine. The repo is present and has the correct baseurl.

```
{'baseurl': 'https://pulp.stage.devshift.net/api/pulp-content/copr/frostyx/test-pulp-dependency/fedora-40-x86_64/',
 'id': 'copr_frostyx_test_pulp_dependency_priority_50',
 'name': 'Additional repo copr_frostyx_test_pulp_dependency_priority_50',
 'priority': 50},
```

And the actual dependency is intalled from it just fine.

```
 python3-copr                         noarch 2.0-1.fc40                 copr_frostyx_test_pulp_dependency_priority_50 243.5 KiB
```

PTAL

File: frontend/coprs_frontend/tests/test_helpers.py [hunk 1]
```diff
@@ -38,62 +34,6 @@ def test_guess_package_name(self):
...
-            dict(args=(m4, http_url),
-                 expected="http://example.com/path/centos-stream-$releasever-$basearch/"),
-            dict(args=(m4, https_url),
-                 expected="https://example.com/path/centos-stream-$releasever-$basearch/")])
-
-        app.config["USE_HTTPS_FOR_RESULTS"] = True
-        for test_set in test_sets:
-            result = generate_repo_url(*test_set["args"])
-            assert result == test_set["expected"]
-
```
[the remark this answers is not in the document]
Comment (FrostyX) [reply frontend/coprs_frontend/tests/test_helpers.py:96]: >  was this intentional?

It most definitely wasn't. I forgot to commit the new file ... again :-(
Updated, please take a look at `test_repos.py` now.

File: frontend/coprs_frontend/coprs/repos.py [hunk 2]
```diff
@@ -71,10 +73,14 @@ def pre_process_repo_url(chroot, repo_url):
...
         user = parsed_url.netloc
         prj = parsed_url.path.split("/")[1]
-        repo_url = "/".join([
-            flask.current_app.config["BACKEND_BASE_URL"],
-            "results", user, prj, chroot
-        ]) + "/"
+        try:
+            copr = CoprsLogic.get_by_ownername_coprname(user, prj)
+            repo_url = "{0}/{1}/".format(copr.repo_url, chroot)
+        except ObjectNotFound:
```
[the remark this answers is not in the document]
Comment (FrostyX) [reply frontend/coprs_frontend/coprs/repos.py:79]: > I thought copr.repo_url takes care of this pulp or backend scenario

Right, this should cover both storages

> why this fallback?

My guess is that people will have random, non-existing `copr://` projects as dependencies. And I didn't want to throw a traceback for them, but rather return the same value that we've always returned.

[the remark this answers is not in the document]
Comment (nikromen) [reply frontend/coprs_frontend/coprs/repos.py:79]: aha, make sense!

File: frontend/coprs_frontend/coprs/repos.py [hunk 3]
```diff
@@ -71,10 +73,14 @@ def pre_process_repo_url(chroot, repo_url):
     if parsed_url.scheme == "copr":
         user = parsed_url.netloc
         prj = parsed_url.path.split("/")[1]
-        repo_url = "/".join([
-            flask.current_app.config["BACKEND_BASE_URL"],
-            "results", user, prj, chroot
-        ]) + "/"
+        try:
+            copr = CoprsLogic.get_by_ownername_coprname(user, prj)
+            repo_url = "{0}/{1}/".format(copr.repo_url, chroot)
```
Comment (praiskup) [inline frontend/coprs_frontend/coprs/repos.py:78]: Isn't this approach dropping the CoprDir structure?

**Pull request merged**